addons/web/static/src/core/network/http_service.js

```diff
@@ -3,8 +3,11 @@
 import { registry } from "../registry";
 
 function checkResponseStatus(response) {
-    if (response.status === 502) {
-        throw new Error("Failed to fetch");
+    switch (response.status) {
+        case 502:
+            throw new Error("Failed to fetch");
+        case 404:
+            throw new Error("Not found");
```

Comment:
I'm not sure this diff is necessary

Reply:
It should ideally indeed be out of scope of this PR. (or at least in a separate commit which the js-team can find acceptable)